src/main/java/io/reactivex/observers/HasDefaultErrorConsumer.java

```diff
@@ -0,0 +1,20 @@
+package io.reactivex.observers;
+
+import io.reactivex.annotations.Experimental;
+
+/**
+ * An interface that indicates that the implementing type has default implementations for error consumption.
+ *
+ * @since 2.1.4 - experimental
+ */
+@Experimental
+public interface HasDefaultErrorConsumer {
```

Comment:
I think we should find a better name for this interface for several reasons:

1. `HasDefaultErrorConsumer` sounds like marker interface that you need to use with `instanceof` to get boolean result.
1. Interface `HasDefaultErrorConsumer` and method name `hasMissingErrorConsumer()` are inconsistent.
1. Extensibility — do we plan to extend it with other functions to check `onNext`/`onComplete` implementations?

Reply:
I'd definitely get rid of "default" and "missing" from both. The negated method is really confusing. There's already `Has*` types which do similar things so I don't agree that it needs changed.

Reply:
Hm, last sentence dropped from my comment for some reason.

Depending on answer to №3 I have following interface names on my mind `ObserverIntrospection`/`SubscriberIntrospection` or  `ErrorConsumerIntrospection`, hope someone has better suggestions :)

>The negated method is really confusing.

Huh, indeed! Didn't notice that until you pointed

Reply:
I let you people work out the naming.

Reply:
for the method, how about just a simple `isNotImplementing()` or `isThrowing()`?

Assuming the interface itself gets named `HasErrorConsumer`?

Reply:
There definitely shouldn't be negation. `!isNotImplementing()` is about the worst thing in the world to see.

I'm curious to the answer to Artem's 3rd question as well.

Reply:
I'd be ok with making it a more generalized name and starting with this, but we'd have to make a call on whether or not this is something others should actually be able to use or is just considered read-only public, as new additions would be breaking changes

Reply:
You can document the interface isn't for implementation by users. The JDK does this a bunch, or it did before 8.

Reply:
That sounds good to me. Should there be specific interfaces for each type in this case to avoid API mismatches?

Reply:
also for the name: how about `CompositeObserver`? Makes it clear that it's made up of multiple components

Reply:
Started with CompositeObserver, will update the tests in a bit but wanted to get this up for a more concrete proposal.

Reply:
People will most likely mistreat `CompositeObserver` and think of it as of a combination of observers because that's how `CompositeDisposable` and `CompositeException` work.

`ObserverInstrumentation`/`ObserverIntrospection`? Such a name also more clearly delivers the intention of consume-only public API

Reply:
`Composite` is misleading and will show up in content assist when the developer types `Composite`

Reply:
to get to the disposable/exception variant mentioned.

Reply:
I'd call it `LambdaConsumerIntrospection` and `hasCustomOnError()`

Reply:
Good points. The `Introspection` moniker feels off to me, mostly because I consider introspection an act and don't necessarily think of it as a noun/type at first. I don't know what a better word would be though. Updated in bae194c